### PR TITLE
✨Support object-fit transitions for amp-lightbox-gallery

### PR DESCRIPTION
**Pull request by sparhami**
- Use https://github.com/ampproject/animations for animating the image for the lightbox transition.
- Change the transitions to use basic CSS animations.

### CODE REVIEW

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 1]
```diff
@@ -899,103 +876,162 @@ export class AmpLightboxGallery extends AMP.BaseElement {
...
+              enter);
         });
   }
 
+  /**
+   * Gets the duration for how long to animate the transition, based on the
+   * distance and size of the viewport.
+   * @param {!Element} srcImg The img to transition from.
+   * @param {!Element} targetImg The img to transition to.
+   * @return {number} How long to animate for, in milliseconds.
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: Do we still need the `@private` annotation? I notice that it's been left out in a few of your new functions.

Comment (sparhami) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: Done.

File: extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js [hunk 2]
```diff
@@ -899,103 +876,162 @@ export class AmpLightboxGallery extends AMP.BaseElement {
...
+        transitionContainer: transLayer,
+        styleContainer: this.getAmpDoc().getHeadNode(),
+        srcImg,
+        targetImg,
+        srcImgRect: undefined,
+        targetImgRect: undefined,
+        styles: {
+          'animationDuration': `${motionDuration}ms`,
+        },
+        keyframesNamespace: undefined,
```
[the remark this answers is not in the document]
Comment (sparhami) [reply extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js]: The library internally generates a new keyframe name using the namespace for each animation (by adding on an incrementing a counter). This allows it to run multiple of the animations at once without you needing to give it a separate namespace for each animation.

**Comment by rsimha** [comment]
@sparhami @cathyxz I'm in the process of upgrading closure compiler, and am running into some problems due to this PR. Specifically, the `@ampproject/animations` module appears to be common JS, and we will no longer support implicit transformation to ESM via closure. Any chance there's an ESM version of this library that we can use?

**Comment by sparhami** [comment]
> @sparhami @cathyxz I'm in the process of upgrading closure compiler, and am running into some problems due to this PR. Specifically, the `@ampproject/animations` module appears to be common JS, and we will no longer support implicit transformation to ESM via closure. Any chance there's an ESM version of this library that we can use?

It has an esm output that sits alongside the commonjs file. I initially tried to use it, but I had a problem with the existing version of Closure Compiler when trying to use it (it worked for a local gulp serve, but failed when trying to use Closure Compiler, e.g. with `gulp check-types`). I think you can switch over by changing this line:

https://github.com/ampproject/amphtml/blob/e3ac7f039896ab077cd3d44298662d1084686af0/build-system/tasks/compile.js#L265

to do `.mjs` instead of `.js` and everything should just work.

**Comment by rsimha** [comment]
Thanks @sparhami, that seemed to work. Will close the issue I filed once I've tested everything.

**Comment by rsimha** [comment]
Hmm... not so fast. Seeing this error with the single pass build. Any ideas?

**Command:**
```
gulp dist --fortesting --single_pass --pseudo_names
```
**Error:**
```
AssertionError [ERR_ASSERTION]: Target node with node_modules/@ampproject/animations/dist/animations.mjs key should exist
    at TopologicalSort.addEdge (/usr/local/google/home/rsimha/src/amphtml/node_modules/topological-sort/dist/index.js:17:9)
    at /usr/local/google/home/rsimha/src/amphtml/build-system/single-pass.js:326:14
    at Array.forEach (<anonymous>)
    at /usr/local/google/home/rsimha/src/amphtml/build-system/single-pass.js:324:17
    at Array.forEach (<anonymous>)
    at Readable.<anonymous> (/usr/local/google/home/rsimha/src/amphtml/build-system/single-pass.js:323:31)
    at Readable.emit (events.js:187:15)
    at endReadableNT (/usr/local/google/home/rsimha/src/amphtml/node_modules/readable-stream/lib/_stream_readable.js:1010:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

**Comment by rsimha** [comment]
Scratch that, fixed by adding `.mjs` to the the list of extensions known by `single-pass.js`. I'll close https://github.com/ampproject/animations/issues/10.

Thanks for the help!

**Pull request merged**